src/Collections/Sequences/Seq.dfy

```diff
@@ -766,4 +766,43 @@ module Seq {
     }
   }
 
+  /* If the multiset of a is a subset of the multiset of b,
+   * then every item in a is in b.
+   */
+  lemma LemmaMultisetSubsetImpliesSeqMemebership<T>(a: seq<T>, b: seq<T>)
+    requires multiset(a) <= multiset(b)
+    ensures forall i | i in a :: i in b
+  {
+    if |a| == 0 {
+    } else {
+      assert multiset{First(a)} <= multiset(b);
+      assert First(a) in b;
+      assert a == [First(a)] + a[1..];
+      LemmaMultisetSubsetImpliesSeqMemebership(a[1..], b);
+    }
+  }
+
+  /* Every item in the parts to be flattend
+   * is in the flattened seq.
+   */
+  lemma LemmaFlattenMembership<T>(parts: seq<seq<T>>, flat: seq<T>)
+    requires Flatten(parts) == flat
+    ensures forall index
+    | 0 <= index < |parts|
+    :: multiset(parts[index]) <= multiset(flat)
```

Comment:
```suggestion
    ensures forall index | 0 <= index < |parts|
      :: multiset(parts[index]) <= multiset(flat)
```

Easier to read this way I find. And can we likewise indent the other `::` lines below?

Reply:
I can.  However the reason I left it like that is because of logical chaining.
If you want to say lots of things in either `|` or `::` you end up with lots more white space.

```dafny
predicate Foo(s: seq<int>)
  ensures forall index
  |
    && 0 <= index < |s|
    && magic(s)
    && other(index)
  ::
    && multiset(s[index]) <= multiset(index)
    && things(multiset(s))
    && toProve(index)
{}
```
The reason I like the above, is that it is clear that `|` and `::` fold up the the `ensures` clause.
Much like the `else` lines up with the `if`

```dafny
if condition then
  doWork()
else
  otherWork()
```

Reply:
That's fair. I'm not convinced I'd want the multi-line formatting for ALL quantifications, even really simple and short ones, but we can argue about that when we get to the auto-formatter :)